### PR TITLE
moving to getDisplayMedia

### PR DESCRIPTION
**Pull request by xiewenlong1202**
use getDisplayMedia instead of chrome plugin

### CODE REVIEW

File: src/sdk/base/mediastream-factory.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ export class AudioTrackConstraints {
   // eslint-disable-next-line require-jsdoc
   constructor(source) {
     if (!Object.values(MediaFormatModule.AudioSourceInfo)
-        .some((v) => v === source)) {
+      .some((v) => v === source)) {
```
Comment (jianjunz) [inline src/sdk/base/mediastream-factory.js]: Please do not making unnecessary changes.

File: src/sdk/base/mediastream-factory.js [hunk 2]
```diff
@@ -147,146 +147,86 @@ export class MediaStreamFactory {
...
-    if (isVideoConstrainsForScreenCast(constraints) && utils.isChrome()) {
-      if (!constraints.extensionId) {
-        return Promise.reject(new TypeError(
-            'Extension ID must be specified for screen sharing on Chrome.'));
-      }
-      const desktopCaptureSources = ['screen', 'window', 'tab'];
-      if (constraints.audio) {
-        desktopCaptureSources.push('audio');
+
+    // check and convert constraints
```
Comment (jianjunz) [inline src/sdk/base/mediastream-factory.js]: Start with capital letter and end with period.

File: src/sdk/base/mediastream-factory.js [hunk 3]
```diff
@@ -147,146 +147,86 @@ export class MediaStreamFactory {
...
-            constraints.video.source ===
-                MediaFormatModule.VideoSourceInfo.SCREENCAST) {
-          mediaConstraints.video.mediaSource = 'screen';
-        }
-      } else {
-        mediaConstraints.video = constraints.video;
+      if (utils.isFirefox() &&
+        constraints.video.source ===
+        MediaFormatModule.VideoSourceInfo.SCREENCAST) {
+        mediaConstraints.video.mediaSource = 'screen';
```
Comment (jianjunz) [inline src/sdk/base/mediastream-factory.js]: Still needed?

**Comment by jianjunz** [comment]
Fixes #2.

**Comment by jianjunz** [comment]
Please also remove screen-sharing-chrome-extension directory as it is no longer needed.

File: src/sdk/base/mediastream-factory.js (hunk 1 above)
Comment (xiewenlong1202) [reply src/sdk/base/mediastream-factory.js]: accepted, revert the format

File: src/sdk/base/mediastream-factory.js (hunk 2 above)
Comment (xiewenlong1202) [reply src/sdk/base/mediastream-factory.js]: Don't got its meaning

File: src/sdk/base/mediastream-factory.js (hunk 3 above)
Comment (xiewenlong1202) [reply src/sdk/base/mediastream-factory.js]: accepted. no need, delete the block.

File: src/sdk/base/mediastream-factory.js (hunk 1 above)
Comment (jianjunz) [reply src/sdk/base/mediastream-factory.js]: Please remove other unnecessary changes as well.

**Comment by xiewenlong1202** [comment]
comments has been resolved
